### PR TITLE
Allow only organization admins to export application

### LINKED ISSUE #5061: [Enhancement] Only allow admins to export application
Currently, anyone with permission to manage pages and datasources can export an application that includes admins and developers. Only admins are expected to have this access

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> This PR fixes the issue where only admins can have the export application functionality.

Fixes #5061

## Type of change

- Enhancement (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested manually with the following steps:

1. Create 2 dummy users: User1, User2
2. Login to User1 account
3. Create a dummy application
4. Try exporting application (_Success_)
5. Share the application with User2 as a developer
6. Login with User2 account, export option will not be available.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: feature/allow-only-admins-to-export-application 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/pages/Applications/permissionHelpers.tsx | 91.67 **(0.76)** | 100 **(0)** | 50 **(0)** | 90.91 **(0.91)**</details>

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/acl/AppsmithRole.java [hunk 1]
```diff
@@ -20,7 +21,7 @@
     APPLICATION_ADMIN("Application Administrator", "", Set.of(MANAGE_APPLICATIONS)),
     APPLICATION_VIEWER("Application Viewer", "",  Set.of(READ_APPLICATIONS)),
     ORGANIZATION_ADMIN("Administrator", "Can modify all organization settings including editing applications and inviting other users to the organization",
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/acl/AppsmithRole.java:23]: Do we need to also update the textual description of the administrator role?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java [hunk 2]
```diff
@@ -92,17 +94,27 @@
...
                 pluginMap.put(plugin.getId(), plugin.getPackageName());
                 return plugin;
             })
-            .then(applicationService.findById(applicationId, AclPermission.MANAGE_APPLICATIONS))
-            .switchIfEmpty(Mono.error(new AppsmithException(
-                AppsmithError.ACL_NO_RESOURCE_FOUND, FieldName.APPLICATION, applicationId))
-            )
-            .flatMap(application -> {
+            .then(applicationMono)
+            .zipWhen(application -> applicationExportPermitted(application))
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java:109]: This step does not seem to be required. Fetching the application with EXPORT_APPLICATIONS permission would ensure that only the users who have the required permissions would pass that step.

**Review by nayan-rafiq — COMMENTED** [review]
I suggest to delete the created resources after the test is completed. For example, if we are creating an organization, we should delete this when the test is completed i.e. inside `@AfterEach` or `@After` methods

**Comment by abhvsn** [comment]
> I suggest to delete the created resources after the test is completed. For example, if we are creating an organization, we should delete this when the test is completed i.e. inside `@AfterEach` or `@After` methods

We are already using `@DirtiesContext`, this should take care of this requirement, right? Are you suggesting that we should do clean-up manually in `@AfterAll`?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java (hunk 2 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java:109]: My bad, I'll push the changes shortly.

**Comment by nayan-rafiq** [comment]
> > I suggest to delete the created resources after the test is completed. For example, if we are creating an organization, we should delete this when the test is completed i.e. inside `@AfterEach` or `@After` methods
> 
> @nayan-rafiq We are already using `@DirtiesContext`, this should take care of this requirement, right? Are you suggesting that we should do clean-up manually in `@AfterAll`?

If `@DirtiesContext` clears the created resources, then there is no need to do anything else.

**Review by nayan-rafiq — APPROVED** [review]

**Review by trishaanand — APPROVED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**